src/elements/ProgressBar/ProgressBar.js

```diff
@@ -98,27 +97,23 @@ export default class ProgressBar extends React.Component {
 
 		let segmentTimeout = 0;
 
-		segments.map((segment) => {
+		segments.forEach(segment => {
 			setTimeout(() => {
-				if (this.state.interval) {
-					clearInterval(this.state.interval);
-				}
+				clearInterval(this.interval);
 				this.startSegmentBar(segment);
 			}, segmentTimeout);
 			segmentTimeout += segment.duration;
 		});
 	}
 
 	startSegmentBar(segment) {
-		const interval = setInterval(this.timer, (segment.duration / segment.percentage));
-
-		this.setState({ interval });
+		this.interval = setInterval(this.timer, (segment.duration / segment.percentage));
 	}
 
 	render() {
 		return (
 			<div>
-				<div className="ui-progress-bar">
+				<div className="ui-progress-bar" style={{ height: `${this.props.height}px` }}>
```

Comment:
React will automatically convert numbers to pixel values. Going along with the comment up top:

```
style={{height: Number(this.props.height)}}
```